Plugin/src/main/java/dev/rosewood/rosestacker/manager/StackManager.java

```diff
@@ -70,6 +71,30 @@ public void reload() {
             long interval = autosaveFrequency * 20 * 60;
             this.autosaveTask = Bukkit.getScheduler().runTaskTimer(this.rosePlugin, () -> this.saveAllData(false), interval, interval);
         }
+
+        this.stackingBasedOnPerformance = ConfigurationManager.Setting.PERFORMANCE_TPS_TOGGLE.getBoolean();
+        if(this.stackingBasedOnPerformance){
+            Bukkit.getServer().getScheduler().runTaskAsynchronously(rosePlugin, new Runnable() {
+                @Override
+                public void run() {
+                    if(ConfigurationManager.Setting.PERFORMANCE_TPS_TOGGLE.getBoolean()){
+                        Bukkit.getScheduler().runTaskTimerAsynchronously(rosePlugin, () -> {
```

Comment:
You need to assign async repeating tasks to a variable and ensure they are cancelled when the plugin shuts down. See how the autosave task is handled in this class.

Additionally, as it is now, since this never gets cancelled and gets run every time /rs reload gets run, you can end up with multiple of these tasks running at once, that's bad.